### ISSUE TITLE
Replace all the javascript with html+css.

### ISSUE THREAD

**Issue by KOLANICH** [coveooss/json-schema-for-humans#4]
https://github.com/you-dont-need/You-Dont-Need-JavaScript

**Comment by dblanchette** [comment]
This is very cool! It actually started out without Javascript, but we ended up using bootstrap as a time saving measure. If I find the time for it, I'll see what I can do to remove Javascript.

**Comment by dblanchette** [comment]
I spent some time investigating the effort needed to remove all javascript and I realized it is a lot. There are a lot of accordion examples without JS, but not really when there are sections within another.

I was thinking of creating another template without the collapsible sections (everything expanded). It would also make it easier to search using the browser search function, and to add anchor links, something I have been wanting to do for a while.

Users that don't want the Bootstrap/jQuery overhead could use that template instead.

Thoughts?

**Comment by mrspacemankey** [comment]
> I was thinking of creating another template without the collapsible sections (everything expanded). It would also make it easier to search using the browser search function, and to add anchor links, something I have been wanting to do for a while.
> 
> Users that don't want the Bootstrap/jQuery overhead could use that template instead.
> 
> Thoughts?

I think this alternative would be useful. Still, at least a "collapse/expand all" button could still be useful, if easily possible with CSS. Including the HTML in-line in a documentation would benefit from this simple overview.

**Comment by KOLANICH** [comment]
>but not really when there are sections within another.

<details>
<summary>a</summary>
<details>
<summary>b</summary>
c
</details>
d
</details>

**Comment by dblanchette** [comment]
@KOLANICH Thanks for the tip! I am more a Python developer than a Web developer, so please excuse my ignorance :)

I started working on a "flat" version without JS, it's not very beautiful, but it generates and renders very quickly!

It's in this branch: https://github.com/coveooss/json-schema-for-humans/tree/feature/flat-version. Comments more than welcome :)

**Comment by kratsg** [comment]
I have an attempt at it with the release on pypi: https://github.com/scikit-hep/pyhf/commit/b615ea6a35b6b1e7343f7d91c602095016d8dae6

Just doing some modifications to the html. One could probably use beautifulsoup to pull out all the relevant pieces, but it would be much easier for all involved to put all the needed css/javascript at the bottom of the body, and then we can just extract the body into the html file as a post-processing step of sorts.. or just have a config for it.

**Comment by KOLANICH** [comment]
@dblanchette, thank you. Sorry for the delay, I have planned to test it and answer after the test, but was busy enough to forget to test.

It seems like in its current state the tool doesn't use `details` tag.

**Comment by dblanchette** [comment]
@KOLANICH Do you know a way to make anchor links and the "expand" and "collapse" buttons work with details tag without javascript? The current implementation uses Javascript to iterate through the page and expand sections that need to be. I did not find any equivalent way without using some Javascript.

@kratsg Thanks! I'll look into making an option as soon as I can get a schema that works. Right now, `allOf`, `anyOf`, etc. (tabbed sections) do not work. I plan on putting some time into this next weekend.

**Comment by KOLANICH** [comment]
https://github.com/coveooss/json-schema-for-humans/issues/4

**Comment by KOLANICH** [comment]
Unfortunately, if we want to expand by a link we have to fall-back to CSS solution.

Here is one adapted from https://cdpn.io/cristina-silva/fullpage/pyXQrj

```html
<html>
<head></head>
<body>
  
<style>
.toggle {
  margin: 0 auto;
  max-width: 400px;
}

.toggle-label {
  background: #fff;
  cursor: pointer;
}

.toggle-label:before {
  content: "+";
  float: left;
}

.toggle-input:checked ~ .toggle-label:before {
  content: "-";
}

.toggle-input:not(checked):target ~ .toggle-label {
  display: none;
}

.close-href {
  display: none;
}

.toggle-input:not(checked):target ~ .close-href {
  display: inline;
}

.toggle-input {
  display: none;
}

.toggle-input:not(checked) ~ .toggle-content {
  display: none;
}

.toggle-input:checked ~ .toggle-content {
  display: block;
}

.toggle-input:target ~ .toggle-content {
  display: block;
}

</style>

<div class="toggle">  
	<!-- Checkbox toggle -->
	<input type="checkbox" id="expand_aaaa" class="toggle-input"></input>
	<a href="#" class="close-href">-aaaaa</a>
	<label for="expand_aaaa" class="toggle-label">aaaaa</label>

	<!-- Content to toggle -->
	<div class="toggle-content">
		bbbb
	</div>
</div>


<a href="#expand_aaaa"> Expand aaaa </a>

</body>
</html>
```

**Comment by dblanchette** [comment]
I have released the `flat` template with version 0.24.

Just generate with `--config template_name=flat` (or any other way you provide the config).

The template uses no Javascript. It still uses Bootstrap for the CSS at the moment. There are no collapsible sections or tabbed sections, everything is rendered one after the other. I had to do it this way since I could not find a way to make anchor links work when a section is collapsed without using Javascript.

This is far from perfect, but I figured it was better to release something rather than letting this nearly one year old issue pending while I try to find time and motivation to make something better :)

I added a section in the README about how to add your own templates if this one does not suit you, it could be a way to have what you need. Contributions to the new `flat` templates would be very appreciated if you can improve its style without adding back Javascript of course!

**Comment by dblanchette** [comment]
Closed for inactivity and because a solution was implemented (even if not as good as I would wish). Feel free to build a better template if you still need this!
